components/base/kinematicbase/ptgKinematics.go

```diff
@@ -157,21 +159,28 @@ func (ptgk *ptgBaseKinematics) GoToInputs(ctx context.Context, inputs []referenc
 		linVel := r3.Vector{0, trajNode.LinVelMMPS, 0}
 		angVel := r3.Vector{0, 0, rdkutils.RadToDeg(trajNode.AngVelRPS)}
 
-		ptgk.logger.Debugf(
-			"setting velocity to linear %v angular %v and running velocity step for %s",
-			linVel,
-			angVel,
-			timestep,
-		)
-
-		err := ptgk.Base.SetVelocity(
-			ctx,
-			linVel,
-			angVel,
-			nil,
-		)
-		if err != nil {
-			return multierr.Combine(err, ptgk.Base.Stop(ctx, nil))
+		// This should call SetVelocity if:
+		// 1) this is the first iteration of the loop, or
+		// 2) either of the linear or angular velocities has changed
+		if i == 0 || !(linVel.ApproxEqual(lastLinVel) && angVel.ApproxEqual(lastAngVel)) {
+			ptgk.logger.Debugf(
+				"setting velocity to linear %v angular %v and running velocity step for %s",
+				linVel,
+				angVel,
+				timestep,
+			)
+
+			err := ptgk.Base.SetVelocity(
+				ctx,
+				linVel,
+				angVel,
+				nil,
+			)
+			if err != nil {
+				return multierr.Combine(err, ptgk.Base.Stop(ctx, nil))
```

Comment:
What happens if you do `multierr.Combine(err,nil)`?
Should we test for whether Stop() returned an error before combining it with the setvelocity error?

Reply:
`multierr.Combine(err,nil)` should result in `err`, or `nil` if `err` is also `nil`.

If we Stop and that returns an error, what would we do aside from just return it?

Reply:
This is good in that case. I was just not sure if `multierr.Combine(err,nil)` might result in a multierr with a nil pointer inside of it which might panic when if the Error method is called on it. 

However it sounds like that isn't something we need to worry about.